File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.py
class Solution:
    def minOperations(self, nums: List[int], numsDivide: List[int]) -> int:
        res=reduce(math.gcd,numsDivide)
        nums.sort()
        dele=0
        for i in nums:
            if(res%i==0):
                return dele
            else:
                dele+=1
        if(dele==len(nums)):
            return -1 

        

File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, vector<int>& numsDivide) {
       int g=numsDivide[0];
       for(int i:numsDivide){
           g= gcd(i,g);
       }
       sort(nums.begin(),nums.end());
       int dele=0;
       for(auto i:nums){
        if(g%i==0){
            return dele;
        }
        else
        dele++;
       }
       return -1;
    }
};